[Issue]: C++ compilation issues during dockerized rocPRIM test build

### Problem Description

Currently running into these build issues in TheRock while enabling tests for rocPRIM in the Docker image during CD/CI builds. There seems to be very generic syntax issues related to the test files in rocPRIM like so:

`/TheRock/math-libs/rocPRIM/test/rocprim/test_invoke_result.cpp:113:1: error: use of undeclared identifier 'gtest_type_params_RocprimInvokeResultUnOpTests_'`

`TheRock/math-libs/rocPRIM/test/rocprim/test_device_search.cpp:110:1: error: a type specifier is required for all declarations`

`error: too many template arguments for class template 'Types'`

i'm continuing to investigate, but wonder if anyone has any insight into these issues? maybe some google-test config i am missing? maybe some c++ issue going strange?

github PR: [Build `prim_test` artifact by marbre · Pull Request #164 · ROCm/TheRock](https://github.com/ROCm/TheRock/pull/164)
logs: [Build `prim_test` artifact · ROCm/TheRock@43b5fda](https://github.com/ROCm/TheRock/actions/runs/13773053344/job/38515906882?pr=164)

### Operating System

AlmaLinux 8.9 (Midnight Oncilla)

### CPU

AMD EPYC 9454 48-Core Processor

### GPU

AMD Instinct MI300X

### ROCm Version

during TheRock build

### ROCm Component

_No response_

### Steps to Reproduce

1. Use docker image as base: ghcr.io/rocm/therock_build_manylinux_x86_64:main

2. use TheRock repo, `prim-tests` branch

3. run the build

### (Optional for Linux users) Output of /opt/rocm/bin/rocminfo --support

_No response_

### Additional Information

_No response_

--- a/math-libs/CMakeLists.txt
+++ b/math-libs/CMakeLists.txt
@@ -78,6 +78,7 @@ if(THEROCK_ENABLE_PRIM)
       -DHIP_PLATFORM=amd
       -DROCM_PATH=
       -DROCM_DIR=
+      -DBUILD_TEST=ON
     COMPILER_TOOLCHAIN
       amd-hip
     RUNTIME_DEPS
@@ -97,6 +98,7 @@ if(THEROCK_ENABLE_PRIM)
       -DHIP_PLATFORM=amd
       -DROCM_PATH=
       -DROCM_DIR=
+      -DBUILD_TEST=ON
     COMPILER_TOOLCHAIN
       amd-hip
     RUNTIME_DEPS
@@ -118,6 +120,7 @@ if(THEROCK_ENABLE_PRIM)
       -DHIP_PLATFORM=amd
       -DROCM_PATH=
       -DROCM_DIR=
+      -DBUILD_TEST=ON
     COMPILER_TOOLCHAIN
       amd-hip
     RUNTIME_DEPS
@@ -137,6 +140,7 @@ if(THEROCK_ENABLE_PRIM)
     COMPONENTS
       dev
       doc
+      test
     SUBPROJECT_DEPS rocPRIM hipCUB rocThrust
   )
 endif(THEROCK_ENABLE_PRIM)

--- a/math-libs/artifact-prim.toml
+++ b/math-libs/artifact-prim.toml
@@ -1,11 +1,27 @@
 # rocPRIM
 [components.dev."math-libs/rocPRIM/stage"]
 [components.doc."math-libs/rocPRIM/stage"]
+[components.test."math-libs/rocPRIM/stage"]
+include = [
+  "bin/test_*",
+  "bin/rocprim/CTestTestfile.cmake",
+]
 
 # hipCUB
 [components.dev."math-libs/hipCUB/stage"]
 [components.doc."math-libs/hipCUB/stage"]
+[components.test."math-libs/hipCUB/stage"]
+include = [
+  "bin/test_*",
+  "bin/hipcub/CTestTestfile.cmake",
+]
 
 # rocThrust
 [components.dev."math-libs/rocThrust/stage"]
 [components.doc."math-libs/rocThrust/stage"]
+[components.test."math-libs/rocThrust/stage"]
+include = [
+  "bin/test_*",
+  "bin/*.hip",
+  "bin/rocthrust/CTestTestfile.cmake",
+]

